dynamic-router/dynamic-router.bal

```diff
@@ -0,0 +1,46 @@
+import ballerina/http;
+import ballerina/mqtt;
+
+type RoutingEntry record {|
+    string deviceId;
+    int roomNo;
+    boolean dimmable;
+|};
+
+type SwitchRequest record {|
+    int roomNo;
+    State state;
+|};
+
+enum State {
+    ON,
+    OFF,
+    DIM
+}
+
+final http:Client deviceClient = check new ("http://api.devices.com.balmock.io");
+map<RoutingEntry> routingTable = {};
```

Comment:
what if we use a table data structure instead of a map? We can mark the `roomNo` attribute to be readonly in the `RoutingEntry` record type.

Reply:
+1 , will update